core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -897,67 +907,77 @@ def _update_pred_proba_dict_with_val_cache(self, model_set: set, model_pred_prob
         model_set = model_set.difference(set(model_pred_proba_dict.keys()))
         return model_set, model_pred_proba_dict
 
-    # TODO: Remove _get_inputs_to_stacker_legacy eventually, move logic internally into this function instead
-    def get_inputs_to_stacker(self, X, base_models, model_pred_proba_dict=None, fit=False, use_orig_features=True, use_val_cache=False):
-        if base_models is None:
-            base_models = []
-        if not fit:
+    def get_inputs_to_stacker(self,
+                              X: pd.DataFrame,
+                              base_models: List[str],
+                              model_pred_proba_dict: dict = None,
+                              fit: bool = False,
+                              use_orig_features: bool = True,
+                              use_val_cache: bool = False) -> pd.DataFrame:
+        """
+        Returns the valid X input for a stacker model with base models equal to `base_models`.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Input data to augment.
+        base_models : List[str]
+            The list of base models to augment X with.
+            Base models will add their prediction probabilities as extra features to X.
+        model_pred_proba_dict : dict, optional
```

Comment:
If this is optional, you should change the type hint to `model_pred_proba_dict: Optional[dict] = None`

Reply:
added